cargo-near-build/src/types/near/build/input/mod.rs

```diff
@@ -47,6 +47,13 @@ pub struct Opts {
     /// additional environment key-value pairs, that should be passed to underlying
     /// build commands
     pub env: Vec<(String, String)>,
+    /// additional environment key-value pairs, that should be passed to underlying
+    /// build commands
+    ///
+    /// unlike [Self::env] this env is *mute*: it's not reflected in `--env` flag-components of environment
+    /// [crate::env_keys::nep330::BUILD_COMMAND] variable,
+    /// set during build
+    pub mute_env: Vec<(String, String)>,
```

Comment:
I find this naming unintuitive. My suggestions:

* nep330_env
* near_env ("near" is a regular word and makes it weird as well)
* implicit_env
* cargo_near_env

Also, would it be harmful if we pass these env variables via `env` as well instead of creating a separate flow for them?